### ISSUE TITLE
SQLite: AUTOINCREMENT not supported

### ISSUE THREAD

**Issue by srittau** [sqlfluff/sqlfluff#3524]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Please consider the following SQLite file, called `foo.sql`:

```sql
CREATE TABLE foo(
    id INTEGER PRIMARY KEY AUTOINCREMENT
);
```

Parsing (or linting) this file with SQLFluff fails due to the `AUTOINCREMENT` flag:

```
$ sqlfluff parse --dialect=sqlite foo.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'StatementSegment'
[L:  1, P:  1]      |            raw:                                              'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            raw:                                              'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            raw:                                              'foo'
[L:  1, P: 17]      |            bracketed:
[L:  1, P: 17]      |                start_bracket:                                '('
[L:  1, P: 18]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                raw:                                          'id'
[L:  2, P:  7]      |                whitespace:                                   ' '
[L:  2, P:  8]      |                raw:                                          'INTEGER'
[L:  2, P: 15]      |                whitespace:                                   ' '
[L:  2, P: 16]      |                raw:                                          'PRIMARY'
[L:  2, P: 23]      |                whitespace:                                   ' '
[L:  2, P: 24]      |                raw:                                          'KEY'
[L:  2, P: 27]      |                whitespace:                                   ' '
[L:  2, P: 28]      |                raw:                                          'AUTOINCREMENT'
[L:  2, P: 41]      |                newline:                                      '\n'
[L:  3, P:  1]      |                end_bracket:                                  ')'
[L:  3, P:  2]      |    statement_terminator:                                     ';'
[L:  3, P:  3]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE foo(\n
                       | id INTEGER PRIMARY...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
```

### Expected Behaviour

I would hope that SQLFluff would support the `AUTOINCREMENT` keyword for SQLite as described [here](https://www.sqlite.org/autoinc.html). If I understand the documentation correctly, only exactly `INTEGER PRIMARY KEY AUTOINCREMENT` (minus whitespace and case) is accepted, but I didn't test that.

### Observed Behaviour

See above. Removing `AUTOINCREMENT` from the example file lets `SQLFluff` parse and lint the file correctly.

### How to reproduce

See above.

### Dialect

SQLite

### Version

```
$ sqlfluff --version
sqlfluff, version 1.0.0
$ python3 --version
Python 3.10.5
```

### Configuration

No extra configuration.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by srittau** [comment]
Thank you for the incredibly fast fix!
